build.sbt

```diff
@@ -21,30 +22,50 @@ lazy val commonSettings = Seq(
   cancelable := true
 )
 
-lazy val bench = crossProject(JVMPlatform, NativePlatform)
+lazy val benchSettings = commonSettings ++ Seq(
+  buildInfoPackage := "rsc.bench",
+  buildInfoUsePackageAsPath := true,
+  buildInfoKeys := Seq[BuildInfoKey](
+    "sourceRoot" -> (baseDirectory in ThisBuild).value
+  )
+)
+
+lazy val benchJavac18 = project
+  .in(file("bench/javac18"))
+  .enablePlugins(BuildInfoPlugin)
+  .enablePlugins(JmhPlugin)
+  .settings(benchSettings)
+
+lazy val benchRsc = crossProject(JVMPlatform, NativePlatform)
   .crossType(CrossType.Full)
-  .in(file("bench"))
-  .dependsOn(rsc, tests)
+  .in(file("bench/rsc"))
+  .dependsOn(rsc)
+  .enablePlugins(BuildInfoPlugin)
   .enablePlugins(JmhPlugin)
+  .jvmSettings(benchCliRscNative("Typecheck"))
```

Comment:
You can resolve https://github.com/twitter/reasonable-scala/issues/22 in the pr while you're at it by adding:
```
.nativeSettings(nativeMode := "release)
```

Reply:
https://github.com/twitter/reasonable-scala/pull/27/commits/263273d91ce3c1e076d13c8f6c70f8658c158930